File: php/src/app/dominio/estados/EstadoConcretoB.php
<?php


class EstadoConcretoB implements EstadoContexto {

    public function actuar(Contexto $contexto) {
        // TODO: Implement actuar() method.
        if($this->cumpleCondicionParaCambiarDeEstado($contexto)){
            //cambiar a otro estado
        }
    }

    private function cumpleCondicionParaCambiarDeEstado(Contexto $contexto) : boolean {
        // TODO
        return true;
    }
}

File: java/src/main/java/dominio/estados/EstadoConcretoB.java
package dominio.estados;

import dominio.Contexto;

public class EstadoConcretoB implements EstadoContexto {

    public void actuar(Contexto contexto) {
        //TODO
        if(cumpleCondicionParaCambiarDeEstado(contexto)){
            //cambiar a otro estado
        }
    }

    private boolean cumpleCondicionParaCambiarDeEstado(Contexto contexto){
        //TODO
        return true;
    }
}